### PR TITLE
Improve pad support

### PR DESCRIPTION
**Pull request by SEt-t**
Two improvements in pad support:

1) Current DS3 code never worked for my original but quite old (made in 2011) DualShock3 in Win7.

It might be an issue with different OS as the same problem was mentioned here: in https://github.com/RPCS3/rpcs3/pull/5933#issuecomment-490246840

In my system Sony's driver (from the wiki's link) doesn't respond at all to Report ID 0xF2 but I can get exactly the same info from Report ID 0. The proposed code adds fallback to ID 0 if ID 0xF2 failed.

2) It's possible to transparently add support for getting accel/gyro data over SCP interface.
It also requires corresponding change in XInput1_3.dll to send the new data, but old one will continue to work and won't break.

### CODE REVIEW

File: rpcs3/Input/xinput_pad_handler.cpp [hunk 1]
```diff
@@ -419,7 +419,7 @@ PadHandlerBase::connection xinput_pad_handler::update_connection(const std::shar
 		return connection::disconnected;
 
 	dev->state = ERROR_NOT_CONNECTED;
-	dev->state_scp = {};
+	dev->state_scp = {.SCP_ACCEL_Y = -0.21875f};
```
Comment (Megamouse) [inline rpcs3/Input/xinput_pad_handler.cpp:422]: what exactly is this value?
And why is it defaulted here

Comment (SEt-t) [reply rpcs3/Input/xinput_pad_handler.cpp:422]: Most values in default state in SCP are represented as 0.0, which corresponds to 512 for accelerometers. Y accelerometer in "neutral" state reads as 400 on my pad, which corresponds to -0.21875 SCP.

It's defaulted here so if SCP interface uses unmodified XInput1_3.dll then fields SCP_ACCEL/GYRO won't be changed and "neutral" state is transparently returned for accelerometers.

Comment (Megamouse) [reply rpcs3/Input/xinput_pad_handler.cpp:422]: The default is 399 based on all other code.
And you should add a comment for that at least.

I imagine this wouldn't be an issue if this accelerometer support could be checked somehow in your modified version.
You could just skip the assignment if it wasn't supported.

**Comment by Megamouse** [comment]
hmm I think something went wrong here...

File: rpcs3/Input/xinput_pad_handler.cpp [hunk 2]
```diff
@@ -450,6 +453,15 @@ void xinput_pad_handler::get_extended_info(const std::shared_ptr<PadDevice>& dev
...
 	pad->m_battery_level = pad->m_cable_state ? BATTERY_LEVEL_FULL : battery_info.BatteryLevel;
+
+	SCP_DS3_ACCEL sensors;
+	if (xinputGetCustomData && xinputGetCustomData(dev->deviceNumber, 0, &sensors) == ERROR_SUCCESS)
+	{
+		pad->m_sensors[0].m_value = sensors.SCP_ACCEL_X;
+		pad->m_sensors[1].m_value = sensors.SCP_ACCEL_Y;
+		pad->m_sensors[2].m_value = sensors.SCP_ACCEL_Z;
+		pad->m_sensors[3].m_value = sensors.SCP_GYRO;
+	}
```
Comment (Megamouse) [inline rpcs3/Input/xinput_pad_handler.cpp:464]:
```suggestion
	if (xinputGetCustomData)
	{
		SCP_DS3_ACCEL sensors;
		if (xinputGetCustomData(dev->deviceNumber, 0, &sensors) == ERROR_SUCCESS)
		{
			pad->m_sensors[0].m_value = sensors.SCP_ACCEL_X;
			pad->m_sensors[1].m_value = sensors.SCP_ACCEL_Y;
			pad->m_sensors[2].m_value = sensors.SCP_ACCEL_Z;
			pad->m_sensors[3].m_value = sensors.SCP_GYRO;
		}
	}
```

**Pull request merged**